src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas1/Cas1SpaceBookingTest.kt

```diff
@@ -789,4 +794,238 @@ class Cas1SpaceBookingTest {
+          ),
+        )
+        .exchange()
+        .expectStatus()
+        .isBadRequest
+        .expectBody()
+        .jsonPath("invalid-params[0].propertyName").isEqualTo("$.premisesId")
+        .jsonPath("invalid-params[0].errorType").isEqualTo("doesNotExist")
+    }
+
+    @Test
+    fun `Returns 400 Bad Request if the spaceBooking id supplied has no matching booking record `() {
+      val (_, jwt) = `Given a User`(roles = listOf(CAS1_FUTURE_MANAGER))
+
+      val unknownSpaceBookingId = UUID.randomUUID()
+
+      webTestClient.post()
+        .uri("/cas1/premises/${premises.id}/space-bookings/$unknownSpaceBookingId/arrival")
+        .header("Authorization", "Bearer $jwt")
+        .bodyValue(
+          Cas1NewArrival(
+            expectedDepartureDate = LocalDate.now().plusMonths(1),
+            arrivalDateTime = LocalDateTime.now().toInstant(ZoneOffset.UTC),
+          ),
+        )
+        .exchange()
+        .expectStatus()
+        .isBadRequest
+        .expectBody()
+        .jsonPath("invalid-params[0].propertyName").isEqualTo("$.bookingId")
+        .jsonPath("invalid-params[0].errorType").isEqualTo("doesNotExist")
+    }
+
+    @Test
+    fun `Returns 400 Bad Request if the booking already has an actual arrival date`() {
+      val (_, jwt) = `Given a User`(roles = listOf(CAS1_FUTURE_MANAGER))
+
+      val (user) = `Given a User`()
+      val (offender) = `Given an Offender`()
+      val (placementRequest) = `Given a Placement Request`(
+        placementRequestAllocatedTo = user,
+        assessmentAllocatedTo = user,
+        createdByUser = user,
+      )
+
+      val spaceBookingWithArrivalDate = cas1SpaceBookingEntityFactory.produceAndPersist {
+        withCrn(offender.otherIds.crn)
+        withPremises(premises)
+        withPlacementRequest(placementRequest)
+        withApplication(placementRequest.application)
+        withCreatedBy(user)
+        withActualArrivalDateTime(Instant.now())
+        withCanonicalArrivalDate(LocalDate.parse("2029-05-29"))
+        withCanonicalDepartureDate(LocalDate.parse("2029-06-29"))
+      }
+
+      webTestClient.post()
+        .uri("/cas1/premises/${premises.id}/space-bookings/${spaceBookingWithArrivalDate.id}/arrival")
+        .header("Authorization", "Bearer $jwt")
+        .bodyValue(
+          Cas1NewArrival(
+            expectedDepartureDate = LocalDate.now().plusMonths(1),
+            arrivalDateTime = LocalDateTime.now().toInstant(ZoneOffset.UTC),
+          ),
+        )
+        .exchange()
+        .expectStatus()
+        .is4xxClientError
+        .expectBody()
+        .jsonPath("title").isEqualTo("Conflict")
+        .jsonPath("status").isEqualTo(409)
+        .jsonPath("detail")
+        .isEqualTo("An arrival is already recorded for this Space Booking: ${spaceBookingWithArrivalDate.id}")
+    }
+
+    @Test
+    fun `Returns 500 Internal Server Error if unexpected failure occurs - invalid key worker )`() {
```

Comment:
i'm going to create a ticket to work what to do when this happens as i feel like it shouldn't necessarily block the activity.